frontend/src/components/pages/secrets/secrets-store-page.tsx

```diff
@@ -0,0 +1,242 @@
+import { proto3 } from '@bufbuild/protobuf';
+import {
+  Badge,
+  Box,
+  Button,
+  ButtonGroup,
+  DataTable,
+  Flex,
+  HStack,
+  Heading,
+  Icon,
+  SearchField,
+  Spinner,
+  Stack,
+  Text,
+  VStack,
+  useDisclosure,
+} from '@redpanda-data/ui';
+import { runInAction } from 'mobx';
+import { Scope } from 'protogen/redpanda/api/dataplane/v1/secret_pb';
+import { useEffect, useState } from 'react';
+import { AiOutlineDelete, AiOutlineEdit } from 'react-icons/ai';
+import { useListSecretsQuery } from 'react-query/api/secret';
+import { uiState } from 'state/uiState';
+import { CreateSecretModal } from './create-secret-modal';
+import { DeleteSecretModal } from './delete-secret-modal';
+import { UpdateSecretModal } from './update-secret-modal';
+
+// Hack for MobX to ensure we don't need to use observables
+export const updatePageTitle = () => {
+  runInAction(() => {
```

Comment:
nit: I'd still prefer to stay consistent with the upper-level mobx classes to handle the logic and remove it with MobX later on. I think we would benefit from consistency.

Reply:
Agree, I am trying to move quite fast, so I had to resort to this to ensure the title is set properly, otherwise it was overriding it.

The problem I have is that with MobX decorators and class-based components or observer pattern I will never get out of the legacy/tech debt we have today, so I decided this is the best we can do for now.